### PR TITLE
Now checks for valid data

### PR DESCRIPTION
**Pull request by Irock23**
This should stop invalid addresses from being added. I haven't actually tested it, but it shouldn't require much modification if any to make it work.

I had it restrict the ports to the range 1024-65535, but it could technically allow any valid port, I suppose.

### CODE REVIEW

**Comment by ghost** [comment]
I want to merge this but I'm afraid it might break and I haven't touched this project. Can anyone test?

**Comment by Shockfire** [comment]
Seems like it should be fine. The regex he's using to validate the IP looks OK. Going to merge.

File: node/index.js [hunk 1]
```diff
@@ -78,24 +78,33 @@ function jsonGet(options, callback) {
...
+    
+    if (!req.query.port) {
         return res.send({result: {code: 1, msg: "Invalid parameters, valid parameters are 'port' (int) and 'shutdown' (bool)"}});
     }
+    var serverPort = +req.query.port;
+    if (isNaN(serverPort) || serverPort < 1024 || serverPort > 65535) {
+      return res.send({result: {code: 4, msg: "Invalid port. A valid port is in the range 1024-65535."}}); //could allow 1-65535
+    }
 
     var ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
```
Comment (dark-c0de) [inline node/index.js:91]: just FYI, this can be spoofed.

Comment (Irock23) [inline node/index.js:91]: The port? Sure, someone can give it any port, but the point is to ensure that the supplied port is an integer in a valid range. We really just don't want someone like me adding "?callback=?" to the end of the port, potentially forcing jQuery to get the server information as JSONP.

**Comment by dark-c0de** [comment]
hmm.. GitHub selected port, I meant to select the ip. X-forwarded-for is not reliable.

**Comment by Irock23** [comment]
IP address can be spoofed without X-Forwarded-For as well. There's not really much we can do about that.

**Pull request merged**